XMonad/Util/WorkspaceCompare.hs

```diff
@@ -64,28 +63,22 @@ getWsCompareByTag = return compare
 --   and screen id. It produces the same ordering as
 --   'XMonad.Hooks.DynamicLog.pprWindowSetXinerama'.
 getXineramaWsCompare :: X WorkspaceCompare
-getXineramaWsCompare = getXineramaWsCompare' False
+getXineramaWsCompare = getXineramaPhysicalWsCompare $ screenComparatorById compare
 
 -- | A comparison function like 'getXineramaWsCompare', but uses physical locations for screens.
-getXineramaPhysicalWsCompare :: X WorkspaceCompare
-getXineramaPhysicalWsCompare = getXineramaWsCompare' True
-
-getXineramaWsCompare' :: Bool -> X WorkspaceCompare
-getXineramaWsCompare' phy = do
+getXineramaPhysicalWsCompare :: ScreenComparator -> X WorkspaceCompare
+getXineramaPhysicalWsCompare (ScreenComparator sc) = do
```

Comment:
You talked about changing the name of this from `Physical` to `Custom`.  I don't think it matters all that much but you could if you wanted.  Leaving in the ability to compare by ID seems fine.